src/app/presentation/components/user-profile/user-profile.component.ts

```diff
@@ -32,6 +49,40 @@ export class UserProfileComponent implements OnInit {
   }
 
   update(){
-    //there will be logic for update user info.
+    if(this.userProfileForm.valid){
+      this.changeUserInfo.Name = this.userProfileForm.get('name')?.value;
+      this.changeUserInfo.Surname = this.userProfileForm.get('surname')?.value;
+      this.changeUserInfo.UserName = this.userProfileForm.get('username')?.value;
+      this.userService.updateUserInfo(this.changeUserInfo).subscribe(
+       () => {
+         Swal.fire({
+           position: 'top-end',
+           icon: 'success',
+           title: 'Update Personal Information',
+           text: "Success",
+           showConfirmButton: false,
+           timer: 1000
+         });
+       },
+       err => {
+         let errorMessage: string = '';
+         if(err.error.errors && typeof err.error.errors === 'object'){
+           const errors = err.error.errors;
+           for(let key in errors){
+             for(let indexError in errors[key]){
+               errorMessage += errors[key][indexError] + '\n';
+             }
+           }
+           this.showAlert(errorMessage);
+           return;
+         } 
+         if(err.error && typeof err.error === 'object'){
+           errorMessage += err.error.error;
+           this.showAlert(errorMessage);
+           return;
```

Comment:
Create tech dept ticket for move it to one place

Reply:
https://app.clickup.com/t/1q4r0q0